zokrates_core/src/flatten/mod.rs

```diff
@@ -2183,34 +2183,57 @@ impl<'ast, T: Field> Flattener<'ast, T> {
         }
     }
 
+    /// Flattens an equality expression
+    ///
+    /// # Arguments
+    ///
+    /// * `statements_flattened` - `FlatStatements<T>` Vector where new flattened statements can be added.
+    /// * `lhs` - `FlatExpression<T>` Left-hand side of the equality expression.
+    /// * `rhs` - `FlatExpression<T>` Right-hand side of the equality expression.
     fn flatten_equality(
```

Comment:
Would be nice to have some tests for these in this file

x == y -> x * 1 = y
x * y == z -> same
z == x * y -> mirror
x * y == z * t -> introduce var

There are similar tests already, this can be written with `flatten_function`

Reply:
Agreed, I wrote some tests to cover all these cases